File: Lab_1/Lab_1.3/Money.cs
namespace Lab_1._3;

public class Money
{
    private int _500hrn { get; set; }
    private int _200hrn { get; set; }
    private int _100hrn { get; set; }
    private int _50hrn { get; set; }
    private int _20hrn { get; set; }
    private int _10hrn { get; set; }
    private int _5hrn { get; set; }
    private int _2hrn { get; set; }
    private int _1hrn { get; set; }

    private int _50kop { get; set; }
    private int _25kop { get; set; }
    private int _10kop { get; set; }
    private int _5kop { get; set; }
    private int _2kop { get; set; }
    private int _1kop { get; set; }

    public bool Init(int _500hrn, int _200hrn, int _100hrn, int _50hrn, int _20hrn, int _10hrn, int _5hrn, int _2hrn, int _1hrn, int _50kop, int _25kop, int _10kop, int _5kop, int _2kop, int _1kop)
    {
        if (_500hrn >= 0 && _200hrn >= 0 && _100hrn >= 0 && _50hrn >= 0 && _20hrn >= 0 && _10hrn >= 0 && _5hrn >= 0 && _2hrn >= 0 && _1hrn >= 0 && _50kop >= 0 && _25kop >= 0 && _10kop >= 0 && _5kop >= 0 && _2kop >= 0 && _1kop >= 0)
        {
            this._500hrn = _500hrn;
            this._200hrn = _200hrn;
            this._100hrn = _100hrn;
            this._50hrn = _50hrn;
            this._20hrn = _20hrn;
            this._10hrn = _10hrn;
            this._5hrn = _5hrn;
            this._2hrn = _2hrn;
            this._1hrn = _1hrn;
            this._50kop = _50kop;
            this._25kop = _25kop;
            this._10kop = _10kop;
            this._5kop = _5kop;
            this._2kop = _2kop;
            this._1kop = _1kop;
            return true;
        }
        else
        {
            Console.WriteLine("Error: Кількість не може бути від'ємною.");
            return false;
        }

    }
    public void Read()
    {

        int _500hrn = 0;
        int _200hrn = 0;
        int _100hrn = 0;
        int _50hrn = 0;
        int _20hrn = 0;
        int _10hrn = 0;
        int _5hrn = 0;
        int _2hrn = 0;
        int _1hrn = 0;

        int _50kop = 0;
        int _25kop = 0;
        int _10kop = 0;
        int _5kop = 0;
        int _2kop = 0;
        int _1kop = 0;

        bool ifCont = true;
        do
        {
            do
            {

                Console.WriteLine("Кiлькість купюр якого номіналу ви хочете ввести, виберiть зi списку:\n");
                Console.WriteLine($"1. Кількість банкнот по 500 грн.");
                Console.WriteLine($"2. Кількість банкнот по 200 грн.");
                Console.WriteLine($"3. Кількість банкнот по 100 грн.");
                Console.WriteLine($"4. Кількість банкнот по 50 грн.");
                Console.WriteLine($"5. Кількість банкнот по 20 грн.");
                Console.WriteLine($"6. Кількість банкнот по 10 грн.");
                Console.WriteLine($"7. Кількість банкнот по 5 грн.");
                Console.WriteLine($"8. Кількість банкнот по 2 грн.");
                Console.WriteLine($"9. Кількість банкнот по 1 грн.");
                Console.WriteLine($"10. Кількість монет по 50 коп.");
                Console.WriteLine($"11. Кількість монет по 25 коп.");
                Console.WriteLine($"12. Кількість монет по 10 коп.");
                Console.WriteLine($"13. Кількість монет по 5 коп.");
                Console.WriteLine($"14. Кількість монет по 2 коп.");
                Console.WriteLine($"15. Кількість монет по 1 коп.");
                Console.WriteLine($"0. Продовжити :)");

                int choice = Convert.ToInt32(Console.ReadLine());

                switch (choice)
                {
                    case 1:
                        Console.WriteLine("Введіть кількість банкнот по 500 грн.: ");
                        _500hrn = Convert.ToInt32(Console.ReadLine());
                        break;
                    case 2:
                        Console.WriteLine("Введіть кількість банкнот по 200 грн.: ");
                        _200hrn = Convert.ToInt32(Console.ReadLine());
                        break;
                    case 3:
                        Console.WriteLine("Введіть кількість банкнот по 100 грн.: ");
                        _100hrn = Convert.ToInt32(Console.ReadLine());
                        break;
                    case 4:
                        Console.WriteLine("Введіть кількість банкнот по 50 грн.: ");
                        _50hrn = Convert.ToInt32(Console.ReadLine());
                        break;
                    case 5:
                        Console.WriteLine("Введіть кількість банкнот по 20 грн.: ");
                        _20hrn = Convert.ToInt32(Console.ReadLine());
                        break;
                    case 6:
                        Console.WriteLine("Введіть кількість банкнот по 10 грн.: ");
                        _10hrn = Convert.ToInt32(Console.ReadLine());
                        break;
                    case 7:
                        Console.WriteLine("Введіть кількість банкнот по 5 грн.: ");
                        _5hrn = Convert.ToInt32(Console.ReadLine());
                        break;
                    case 8:
                        Console.WriteLine("Введіть кількість банкнот по 2 грн.: ");
                        _2hrn = Convert.ToInt32(Console.ReadLine());
                        break;
                    case 9:
                        Console.WriteLine("Введіть кількість банкнот по 1 грн.: ");
                        _1hrn = Convert.ToInt32(Console.ReadLine());
                        break;
                    case 10:
                        Console.WriteLine("Введіть кількість монет по 50 коп.: ");
                        _50kop = Convert.ToInt32(Console.ReadLine());
                        break;
                    case 11:
                        Console.WriteLine("Введіть кількість монет по 25 коп.: ");
                        _25kop = Convert.ToInt32(Console.ReadLine());
                        break;
                    case 12:
                        Console.WriteLine("Введіть кількість монет по 10 коп.: ");
                        _10kop = Convert.ToInt32(Console.ReadLine());
                        break;
                    case 13:
                        Console.WriteLine("Введіть кількість монет по 5 коп.: ");
                        _5kop = Convert.ToInt32(Console.ReadLine());
                        break;
                    case 14:
                        Console.WriteLine("Введіть кількість монет по 2 коп.: ");
                        _2kop = Convert.ToInt32(Console.ReadLine());
                        break;
                    case 15:
                        Console.WriteLine("Введіть кількість монет по 1 коп.: ");
                        _1kop = Convert.ToInt32(Console.ReadLine());
                        break;
                    case 0:
                        ifCont = false;
                        break;
                    default:
                        Console.WriteLine("Неправильний вибір.");
                        break;
                }
            } while (!Init(_500hrn, _200hrn, _100hrn, _50hrn, _20hrn, _10hrn, _5hrn, _2hrn, _1hrn, _50kop, _25kop, _10kop, _5kop, _2kop, _1kop));
        } while (ifCont);
    }
    public void Display()
    {
        Console.WriteLine("\n");
        Console.WriteLine("Список кількостей банкнот та монет:\n");
        Console.WriteLine($"Кількість банкнот по 500 грн.:  {_500hrn}");
        Console.WriteLine($"Кількість банкнот по 200 грн.:  {_200hrn}");
        Console.WriteLine($"Кількість банкнот по 100 грн.:  {_100hrn}");
        Console.WriteLine($"Кількість банкнот по 50 грн.:   {_50hrn}");
        Console.WriteLine($"Кількість банкнот по 20 грн.:   {_20hrn}");
        Console.WriteLine($"Кількість банкнот по 10 грн.:   {_10hrn}");
        Console.WriteLine($"Кількість банкнот по 5 грн.:    {_5hrn}");
        Console.WriteLine($"Кількість банкнот по 2 грн.:    {_2hrn}");
        Console.WriteLine($"Кількість банкнот по 1 грн.:    {_1hrn}");
        Console.WriteLine($"Кількість монет по 50 коп.:     {_50kop}");
        Console.WriteLine($"Кількість монет по 25 коп.:     {_25kop}");
        Console.WriteLine($"Кількість монет по 10 коп.:     {_10kop}");
        Console.WriteLine($"Кількість монет по 5 коп.:      {_5kop}");
        Console.WriteLine($"Кількість монет по 2 коп.:      {_2kop}");
        Console.WriteLine($"Кількість монет по 1 коп.:      {_1kop}");

    }
    public string toString()
    {
        string result = "";

        result += $"Кількість банкнот по 500 грн.: {_500hrn}\n";
        result += $"Кількість банкнот по 200 грн.: {_200hrn}\n";
        result += $"Кількість банкнот по 100 грн.: {_100hrn}\n";
        result += $"Кількість банкнот по 50 грн.: {_50hrn}\n";
        result += $"Кількість банкнот по 20 грн.: {_20hrn}\n";
        result += $"Кількість банкнот по 10 грн.: {_10hrn}\n";
        result += $"Кількість банкнот по 5 грн.: {_5hrn}\n";
        result += $"Кількість банкнот по 2 грн.: {_2hrn}\n";
        result += $"Кількість банкнот по 1 грн.: {_1hrn}\n";
        result += $"Кількість монет по 50 коп.: {_50kop}\n";
        result += $"Кількість монет по 25 коп.: {_25kop}\n";
        result += $"Кількість монет по 10 коп.: {_10kop}\n";
        result += $"Кількість монет по 5 коп.: {_5kop}\n";
        result += $"Кількість монет по 2 коп.: {_2kop}\n";
        result += $"Кількість монет по 1 коп.: {_1kop}";

        return result;
    }
    public double CalculateTotal()
    {
        double total = 0;

        total += _500hrn * 500;
        total += _200hrn * 200;
        total += _100hrn * 100;
        total += _50hrn * 50;
        total += _20hrn * 20;
        total += _10hrn * 10;
        total += _5hrn * 5;
        total += _2hrn * 2;
        total += _1hrn;

        total += _50kop * 0.5;
        total += _25kop * 0.25;
        total += _10kop * 0.1;
        total += _5kop * 0.05;
        total += _2kop * 0.02;
        total += _1kop * 0.01;

        return total;
    }

    public static Money Suma(Money money1, Money money2)
    {
        Money newMoney = new Money();

        newMoney._500hrn = money1._500hrn + money2._500hrn;
        newMoney._200hrn = money1._200hrn + money2._200hrn;
        newMoney._100hrn = money1._100hrn + money2._100hrn;
        newMoney._50hrn = money1._50hrn + money2._50hrn;
        newMoney._20hrn = money1._20hrn + money2._20hrn;
        newMoney._10hrn = money1._10hrn + money2._10hrn;
        newMoney._5hrn = money1._5hrn + money2._5hrn;
        newMoney._2hrn = money1._2hrn + money2._2hrn;
        newMoney._1hrn = money1._1hrn + money2._1hrn;

        newMoney._50kop = money1._50kop + money2._50kop;
        newMoney._25kop = money1._25kop + money2._25kop;
        newMoney._10kop = money1._10kop + money2._10kop;
        newMoney._5kop = money1._5kop + money2._5kop;
        newMoney._2kop = money1._2kop + money2._2kop;
        newMoney._1kop = money1._1kop + money2._1kop;

        return newMoney;

    }
    public static Money Subtract(Money money1, Money money2)
    {
        Money newMoney = new Money();

        newMoney._500hrn = money1._500hrn - money2._500hrn;
        newMoney._200hrn = money1._200hrn - money2._200hrn;
        newMoney._100hrn = money1._100hrn - money2._100hrn;
        newMoney._50hrn = money1._50hrn - money2._50hrn;
        newMoney._20hrn = money1._20hrn - money2._20hrn;
        newMoney._10hrn = money1._10hrn - money2._10hrn;
        newMoney._5hrn = money1._5hrn - money2._5hrn;
        newMoney._2hrn = money1._2hrn - money2._2hrn;
        newMoney._1hrn = money1._1hrn - money2._1hrn;

        newMoney._50kop = money1._50kop - money2._50kop;
        newMoney._25kop = money1._25kop - money2._25kop;
        newMoney._10kop = money1._10kop - money2._10kop;
        newMoney._5kop = money1._5kop - money2._5kop;
        newMoney._2kop = money1._2kop - money2._2kop;
        newMoney._1kop = money1._1kop - money2._1kop;

        return newMoney;
    }
    public Money MultiplyByDecimal(double multiplier)
    {
        Money newMoney = new()
        {
            _500hrn = (int)(_500hrn * multiplier),
            _200hrn = (int)(_200hrn * multiplier),
            _100hrn = (int)(_100hrn * multiplier),
            _50hrn = (int)(_50hrn * multiplier),
            _20hrn = (int)(_20hrn * multiplier),
            _10hrn = (int)(_10hrn * multiplier),
            _5hrn = (int)(_5hrn * multiplier),
            _2hrn = (int)(_2hrn * multiplier),
            _1hrn = (int)(_1hrn * multiplier),

            _50kop = (int)(_50kop * multiplier),
            _25kop = (int)(_25kop * multiplier),
            _10kop = (int)(_10kop * multiplier),
            _5kop = (int)(_5kop * multiplier),
            _2kop = (int)(_2kop * multiplier),
            _1kop = (int)(_1kop * multiplier)
        };

        return newMoney;
    }
}

File: Lab_2/Lab_2.3_c++/Money.cpp
#include "Money.h"
#include <iostream>
#include <iomanip>
#include <cmath>
#include <stdlib.h>
#include <sstream>

using namespace std;

Money::Money() {
	_500hrn = _200hrn = _100hrn = _50hrn = _20hrn = _10hrn = _5hrn = _2hrn = _1hrn = 0;
	_50kop = _25kop = _10kop = _5kop = _2kop = _1kop = 0;
}
Money::Money(int _500HRN, int _200HRN, int _100HRN, int _50HRN, int _20HRN, int _10HRN, int _5HRN, int _2HRN, int _1HRN,
	int _50KOP, int _25KOP, int _10KOP, int _5KOP, int _2KOP, int _1KOP) {
	if (_500HRN >= 0 || _200HRN >= 0 || _100HRN >= 0 || _50HRN >= 0 || _20HRN >= 0 || _10HRN >= 0 || _5HRN >= 0 || _2HRN >= 0 || _1HRN >= 0 ||
		_50KOP >= 0 || _25KOP >= 0 || _10KOP >= 0 || _5KOP >= 0 || _2KOP >= 0 || _1KOP >= 0) {

			{

				_500hrn = _500HRN;
				_200hrn = _200HRN;
				_100hrn = _100HRN;
				_50hrn = _50HRN;
				_20hrn = _20HRN;
				_10hrn = _10HRN;
				_5hrn = _5HRN;
				_2hrn = _2HRN;
				_1hrn = _1HRN;
				_50kop = _50KOP;
				_25kop = _25KOP;
				_10kop = _10KOP;
				_5kop = _5KOP;
				_2kop = _2KOP;
				_1kop = _1KOP;

			}
	}
}
Money::Money(const Money& m) {
	*this = m;
}
void Money::set500hrn(int value) {
	if (value >= 0) {
		_500hrn = value;
	}
	else {
		cout << "Error: value must be greater than or equal to 0." << endl;
	}
}

void Money::set200hrn(int value) {
	if (value >= 0) {
		_200hrn = value;
	}
	else {
		cout << "Error: value must be greater than or equal to 0." << endl;
	}
}
void Money::set100hrn(int value) {
	if (value >= 0) {
		_100hrn = value;
	}
	else {
		cout << "Error: value must be greater than or equal to 0." << endl;
	}
}

void Money::set50hrn(int value) {
	if (value >= 0) {
		_50hrn = value;
	}
	else {
		cout << "Error: value must be greater than or equal to 0." << endl;
	}
}

void Money::set20hrn(int value) {
	if (value >= 0) {
		_20hrn = value;
	}
	else {
		cout << "Error: value must be greater than or equal to 0." << endl;
	}
}

void Money::set10hrn(int value) {
	if (value >= 0) {
		_10hrn = value;
	}
	else {
		cout << "Error: value must be greater than or equal to 0." << endl;
	}
}

void Money::set5hrn(int value) {
	if (value >= 0) {
		_5hrn = value;
	}
	else {
		cout << "Error: value must be greater than or equal to 0." << endl;
	}
}

void Money::set2hrn(int value) {
	if (value >= 0) {
		_2hrn = value;
	}
	else {
		cout << "Error: value must be greater than or equal to 0." << endl;
	}
}

void Money::set1hrn(int value) {
	if (value >= 0) {
		_1hrn = value;
	}
	else {
		cout << "Error: value must be greater than or equal to 0." << endl;
	}
}

void Money::set50kop(int value) {
	if (value >= 0) {
		_50kop = value;
	}
	else {
		cout << "Error: value must be greater than or equal to 0." << endl;
	}
}

void Money::set25kop(int value) {
	if (value >= 0) {
		_25kop = value;
	}
	else {
		cout << "Error: value must be greater than or equal to 0." << endl;
	}
}

void Money::set10kop(int value) {
	if (value >= 0) {
		_10kop = value;
	}
	else {
		cout << "Error: value must be greater than or equal to 0." << endl;
	}
}

void Money::set5kop(int value) {
	if (value >= 0) {
		_5kop = value;
	}
	else {
		cout << "Error: value must be greater than or equal to 0." << endl;
	}
}

void Money::set2kop(int value) {
	if (value >= 0) {
		_2kop = value;
	}
	else {
		cout << "Error: value must be greater than or equal to 0." << endl;
	}
}

void Money::set1kop(int value) {
	if (value >= 0) {
		_1kop = value;
	}
	else {
		cout << "Error: value must be greater than or equal to 0." << endl;
	}
}

Money& Money::operator++() { // Prefix increment operator
	++_500hrn;
	return *this;
}

Money& Money::operator--() { // Prefix decrement operator
	--_500hrn;
	return *this;
}

Money Money::operator++(int) { // Postfix increment operator
	Money tmp = *this;
	_100hrn++;
	return tmp;
}

Money Money::operator--(int) { // Postfix decrement operator
	Money tmp = *this;
	_100hrn--;
	return tmp;
}
Money& Money::operator=(const Money& other) {
	if (this != &other) {
		_500hrn = other._500hrn;
		_200hrn = other._200hrn;
		_100hrn = other._100hrn;
		_50hrn = other._50hrn;
		_20hrn = other._20hrn;
		_10hrn = other._10hrn;
		_5hrn = other._5hrn;
		_2hrn = other._2hrn;
		_1hrn = other._1hrn;
		_50kop = other._50kop;
		_25kop = other._25kop;
		_10kop = other._10kop;
		_5kop = other._5kop;
		_2kop = other._2kop;
		_1kop = other._1kop;
	}
	return *this;
}


istream& operator >>(istream& in, Money& m)
{
	cout << "Кількість банкнот по 500 грн.: ";
	in >> m._500hrn;
	cout << "Кількість банкнот по 200 грн.: ";
	in >> m._200hrn;
	cout << "Кількість банкнот по 100 грн.: ";
	in >> m._100hrn;
	cout << "Кількість банкнот по 50 грн.: ";
	in >> m._50hrn;
	cout << "Кількість банкнот по 20 грн.: ";
	in >> m._20hrn;
	cout << "Кількість банкнот по 10 грн.: ";
	in >> m._10hrn;
	cout << "Кількість банкнот по 5 грн.: ";
	in >> m._5hrn;
	cout << "Кількість банкнот по 2 грн.: ";
	in >> m._2hrn;
	cout << "Кількість банкнот по 1 грн.: ";
	in >> m._1hrn;
	cout << "Кількість монет по 50 коп.: ";
	in >> m._50kop;
	cout << "Кількість монет по 25 коп.: ";
	in >> m._25kop;
	cout << "Кількість монет по 10 коп.: ";
	in >> m._10kop;
	cout << "Кількість монет по 5 коп.: ";
	in >> m._5kop;
	cout << "Кількість монет по 2 коп.: ";
	in >> m._2kop;
	cout << "Кількість монет по 1 коп.: ";
	in >> m._1kop;
	return in;
}
ostream& operator <<(ostream& out, const Money& m)
{
	out << string(m) << endl;
	return out;
}
Money::operator string() const
{
	stringstream sout;
	sout << "Кількість банкнот по 500 грн.: " << _500hrn << "\n";
	sout << "Кількість банкнот по 200 грн.: " << _200hrn << "\n";
	sout << "Кількість банкнот по 100 грн.: " << _100hrn << "\n";
	sout << "Кількість банкнот по 50 грн.: " << _50hrn << "\n";
	sout << "Кількість банкнот по 20 грн.: " << _20hrn << "\n";
	sout << "Кількість банкнот по 10 грн.: " << _10hrn << "\n";
	sout << "Кількість банкнот по 5 грн.: " << _5hrn << "\n";
	sout << "Кількість банкнот по 2 грн.: " << _2hrn << "\n";
	sout << "Кількість банкнот по 1 грн.: " << _1hrn << "\n";
	sout << "Кількість монет по 50 коп.: " << _50kop << "\n";
	sout << "Кількість монет по 25 коп.: " << _25kop << "\n";
	sout << "Кількість монет по 10 коп.: " << _10kop << "\n";
	sout << "Кількість монет по 5 коп.: " << _5kop << "\n";
	sout << "Кількість монет по 2 коп.: " << _2kop << "\n";
	sout << "Кількість монет по 1 коп.: " << _1kop;
	return sout.str();
}

double Money::CalculateTotal() {
	double total = 0;

	total += _500hrn * 500;
	total += _200hrn * 200;
	total += _100hrn * 100;
	total += _50hrn * 50;
	total += _20hrn * 20;
	total += _10hrn * 10;
	total += _5hrn * 5;
	total += _2hrn * 2;
	total += _1hrn;

	total += _50kop * 0.5;
	total += _25kop * 0.25;
	total += _10kop * 0.1;
	total += _5kop * 0.05;
	total += _2kop * 0.02;
	total += _1kop * 0.01;

	return total;
}


Money Money::Suma(const Money& money1, const Money& money2) {
	Money newMoney;

	newMoney._500hrn = money1._500hrn + money2._500hrn;
	newMoney._200hrn = money1._200hrn + money2._200hrn;
	newMoney._100hrn = money1._100hrn + money2._100hrn;
	newMoney._50hrn = money1._50hrn + money2._50hrn;
	newMoney._20hrn = money1._20hrn + money2._20hrn;
	newMoney._10hrn = money1._10hrn + money2._10hrn;
	newMoney._5hrn = money1._5hrn + money2._5hrn;
	newMoney._2hrn = money1._2hrn + money2._2hrn;
	newMoney._1hrn = money1._1hrn + money2._1hrn;

	newMoney._50kop = money1._50kop + money2._50kop;
	newMoney._25kop = money1._25kop + money2._25kop;
	newMoney._10kop = money1._10kop + money2._10kop;
	newMoney._5kop = money1._5kop + money2._5kop;
	newMoney._2kop = money1._2kop + money2._2kop;
	newMoney._1kop = money1._1kop + money2._1kop;

	return newMoney;
}

Money Money::Subtract(const Money& money1, const Money& money2) {

	Money newMoney;

	newMoney._500hrn = money1._500hrn - money2._500hrn;
	newMoney._200hrn = money1._200hrn - money2._200hrn;
	newMoney._100hrn = money1._100hrn - money2._100hrn;
	newMoney._50hrn = money1._50hrn - money2._50hrn;
	newMoney._20hrn = money1._20hrn - money2._20hrn;
	newMoney._10hrn = money1._10hrn - money2._10hrn;
	newMoney._5hrn = money1._5hrn - money2._5hrn;
	newMoney._2hrn = money1._2hrn - money2._2hrn;
	newMoney._1hrn = money1._1hrn - money2._1hrn;

	newMoney._50kop = money1._50kop - money2._50kop;
	newMoney._25kop = money1._25kop - money2._25kop;
	newMoney._10kop = money1._10kop - money2._10kop;
	newMoney._5kop = money1._5kop - money2._5kop;
	newMoney._2kop = money1._2kop - money2._2kop;
	newMoney._1kop = money1._1kop - money2._1kop;

	return newMoney;
}
Money Money::MultiplyByDecimal(double multiplier) {
	Money newMoney(
		_500hrn = (int)(_500hrn * multiplier),
		_200hrn = (int)(_200hrn * multiplier),
		_100hrn = (int)(_100hrn * multiplier),
		_50hrn = (int)(_50hrn * multiplier),
		_20hrn = (int)(_20hrn * multiplier),
		_10hrn = (int)(_10hrn * multiplier),
		_5hrn = (int)(_5hrn * multiplier),
		_2hrn = (int)(_2hrn * multiplier),
		_1hrn = (int)(_1hrn * multiplier),

		_50kop = (int)(_50kop * multiplier),
		_25kop = (int)(_25kop * multiplier),
		_10kop = (int)(_10kop * multiplier),
		_5kop = (int)(_5kop * multiplier),
		_2kop = (int)(_2kop * multiplier),
		_1kop = (int)(_1kop * multiplier));


	return newMoney;
}
Money operator+(const Money& money1, const Money& money2) {
	Money result;
	result._500hrn = money1._500hrn + money2._500hrn;
	result._200hrn = money1._200hrn + money2._200hrn;
	result._100hrn = money1._100hrn + money2._100hrn;
	result._50hrn = money1._50hrn + money2._50hrn;
	result._20hrn = money1._20hrn + money2._20hrn;
	result._10hrn = money1._10hrn + money2._10hrn;
	result._5hrn = money1._5hrn + money2._5hrn;
	result._2hrn = money1._2hrn + money2._2hrn;
	result._1hrn = money1._1hrn + money2._1hrn;

	result._50kop = money1._50kop + money2._50kop;
	result._25kop = money1._25kop + money2._25kop;
	result._10kop = money1._10kop + money2._10kop;
	result._5kop = money1._5kop + money2._5kop;
	result._2kop = money1._2kop + money2._2kop;
	result._1kop = money1._1kop + money2._1kop;

	return result;
}

Money operator-(const Money& money1, const Money& money2) {
	Money result;
	result._500hrn = money1._500hrn - money2._500hrn;
	result._200hrn = money1._200hrn - money2._200hrn;
	result._100hrn = money1._100hrn - money2._100hrn;
	result._50hrn = money1._50hrn - money2._50hrn;
	result._20hrn = money1._20hrn - money2._20hrn;
	result._10hrn = money1._10hrn - money2._10hrn;
	result._5hrn = money1._5hrn - money2._5hrn;
	result._2hrn = money1._2hrn - money2._2hrn;
	result._1hrn = money1._1hrn - money2._1hrn;

	result._50kop = money1._50kop - money2._50kop;
	result._25kop = money1._25kop - money2._25kop;
	result._10kop = money1._10kop - money2._10kop;
	result._5kop = money1._5kop - money2._5kop;
	result._2kop = money1._2kop - money2._2kop;
	result._1kop = money1._1kop - money2._1kop;

	return result;
}

Money operator*(const Money& money, double multiplier) {
	Money result;
	result._500hrn = static_cast<int>(money._500hrn * multiplier);
	result._200hrn = static_cast<int>(money._200hrn * multiplier);
	result._100hrn = static_cast<int>(money._100hrn * multiplier);
	result._50hrn = static_cast<int>(money._50hrn * multiplier);
	result._20hrn = static_cast<int>(money._20hrn * multiplier);
	result._10hrn = static_cast<int>(money._10hrn * multiplier);
	result._5hrn = static_cast<int>(money._5hrn * multiplier);
	result._2hrn = static_cast<int>(money._2hrn * multiplier);
	result._1hrn = static_cast<int>(money._1hrn * multiplier);

	result._50kop = static_cast<int>(money._50kop * multiplier);
	result._25kop = static_cast<int>(money._25kop * multiplier);
	result._10kop = static_cast<int>(money._10kop * multiplier);
	result._5kop = static_cast<int>(money._5kop * multiplier);
	result._2kop = static_cast<int>(money._2kop * multiplier);
	result._1kop = static_cast<int>(money._1kop * multiplier);

	return result;
}
